pubnub-gson/pubnub-gson-api/src/main/java/com/pubnub/api/java/PubNub.kt

```diff
@@ -463,44 +473,122 @@ interface PubNub : EventEmitter, StatusEmitter {
     /**
      * The method returns a list of channel memberships for a user. This method doesn't return a user's subscriptions.
      */
+    @Deprecated(
+        message = "This function is deprecated. Use getMemberships(userId: String) for better functionality.",
+        replaceWith = ReplaceWith(
+            "getMemberships(userId)"
+        )
+    )
     fun getMemberships(): GetMemberships
 
+    /**
+     * The method returns a list of channel memberships for a user. This method doesn't return a user's subscriptions.
+     */
+    fun getMemberships(userId: String): GetMembershipsBuilder
+
     /**
      * Set channel memberships for a UUID.
      */
-    fun setMemberships(): SetMemberships.Builder
+    @Deprecated(
+        message = "Use setMemberships(Collection<PNChannelMembership>) instead.",
+        replaceWith = ReplaceWith("setMemberships(channelMemberships)")
+    )
+    fun setMemberships(): SetMemberships.Builder // add deprecation
+
+    /**
+     * Set channel memberships for a User.
+     */
+    fun setMemberships(channelMemberships: Collection<PNChannelMembership>): SetMembershipsBuilder
 
     /**
      * Remove channel memberships for a UUID.
      */
+    @Deprecated(
+        message = "Use removeMemberships(Collection<PNChannelMembership>) instead.",
+        replaceWith = ReplaceWith("removeMemberships(channelMemberships)")
+    )
     fun removeMemberships(): RemoveMemberships.Builder
 
     /**
-     * Add and remove channel memberships for a UUID.
+     * Remove channel memberships for a UUID.
+     */
+    fun removeMemberships(channelMemberships: Collection<PNChannelMembership>): RemoveMembershipsBuilder
```

Comment:
Initially I wanted to do this to have it similar to Kotlin SDK but then I thought that maybe that this is not that bad because we can use result from getMembership and feed it to removeMemberships. Now I see that returned data from getMembership and input data to removeMemberships are not the same type (PNMembership vs. PNChannelMembership). In this situation I will change it to Collection<String>